librz/parse/parse.c

```diff
@@ -154,9 +154,18 @@ RZ_API char *rz_parse_immtrim(char *opstr) {
 	return opstr;
 }
 
-RZ_API bool rz_parse_subvar(RzParse *p, RzAnalysisFunction *f, ut64 addr, int oplen, char *data, char *str, int len) {
+/*
+* \brief Substitutes register relative accesses with function variable names
+* \param p The parser
+* \param f The function
+* \param op The analysis op of the current instruction
+* \param data The disassembly of the current instruction
+* \param str The string buffer to write the output to
+* \param len The length of the string buffer
+*/
+RZ_API bool rz_parse_subvar(RZ_BORROW RZ_NONNULL RzParse *p, RZ_BORROW RZ_NULLABLE RzAnalysisFunction *f, RZ_BORROW RZ_NONNULL RzAnalysisOp *op, RZ_BORROW RZ_NONNULL RZ_IN char *data, RZ_BORROW RZ_NONNULL RZ_OUT char *str, int len) {
 	if (p->cur && p->cur->subvar) {
```

Comment:
add an rz_return_if_fail for each RZ_NONNULL, also i do not think is needed to specify `RZ_BORROW` when you pass a variable as input, jusr RZ_IN/RZ_OUT for pointers that are used to return a value (which in this case you do not have)

Reply:
Agree